src/prefect/orion/orchestration/core_policy.py

```diff
@@ -305,6 +292,55 @@ async def before_transition(
         await self.reject_transition(state=retry_state, reason="Retrying")
 
 
+class RestartFlowRun(BaseOrchestrationRule):
+    FROM_STATES = TERMINAL_STATES
+    TO_STATES = [states.StateType.SCHEDULED]
+
+    async def before_transition(
+        self,
+        initial_state: Optional[states.State],
+        proposed_state: Optional[states.State],
+        context: FlowOrchestrationContext,
+    ):
+        if context.run.parent_task_run_id:
+            await self.abort_transition("Cannot restart a subflow run.")
+
+        if not context.run.deployment_id:
+            await self.abort_transition(
+                "Cannot restart a run without an associated deployment."
+            )
+
+        # reset run count to 0
+        self.original_run_count = context.run.run_count
+        context.run.run_count = 0  # reset run count to preserve retry behavior
+
+        # update empirical settings
+        self.original_settings = context.run_settings.copy()
+        self.restarts = self.original_settings.restarts + 1
+        new_settings = context.run_settings.copy()
+        new_settings.restarts = self.restarts
+        flow_run_update = actions.FlowRunUpdate(empirical_policy=new_settings)
```

Comment:
Before merge, we should definitely be tracking this in the state details or on the flow run since empirical settings is intended to be static.